lua/lspsaga/diagnostic.lua

```diff
@@ -15,6 +15,20 @@ M.highlights = {
   [vim.diagnostic.severity.HINT] = "DiagnosticFloatingHint",
 }
 
+M.last_messages = {}
+
+M.yank_last_messages = function(opts)
+  if #M.last_messages == 0 then
+    return
+  end
+  local reg = opts or '"' -- yank to " by default
+  local message = M.last_messages[1]
+  for i = 2, #M.last_messages do
+    message = message .. '\n' .. M.last_messages[i]
+  end
+  vim.fn.setreg(reg, message)
+end
```

Comment:
Did some testing, at first I ran it without calling `show_line_diagnostics`,  which copied nothing "I mean of course". Additionally, it didn't notify me that there is nothing to be copied. 

Therefore, the following needs to be done

- Clear `M.last_messages` when content is copied to registry.
- Get current line diagnostics messages when `M.last_messages` is empty
- if messages still empty, notify nothing to be copied.

With that being said, I believe extracting how messages in `show_diagnostics` get processed is first critical step..

Reply:
Thank you (and sorry for the late reply), working on it.

A few questions tho:

Do you have any suggestions for the subcommand name?

Also, what do you think about having a subcommand that will always copy the current line diagnostics (ignoring the M.last_messages)?

Reply:
> Do you have any suggestions for the subcommand name?

you mean diagnostic_yank_messages?

I think it's okay, you happens to have another suggestion?

> what do you think about having a subcommand that will always copy the current line diagnostics (ignoring the M.last_messages)?
 
make sense, perhaps user ca run something like `diagnostic_yank_messages last +` to always get the last message or `diagnostic_yank_messages +` to ignore M.last_messages totally. Does this look sane to you?

Reply:
I was thinking of something like `yank_line_diagnostics` (to be closer to the `show_line_diagnostics` subcommand) and `yank_last_diagnostics`. What do you think?

Reply:
`yank_line_diagnostics` consistency 😍.

🤔 I believe like show it should just get fresh results from the current line, and don't even consider last_messages?  this or confusing the users with choosing between very similar things.

We're most likely would either show then yank or yank without showing, and performance, I'd bet that both (caching to M.last_messages) and getting new results each time, would have identical wait.

Reply:
(maybe i misunderstood the comment, sorry for my bad english)

> I believe like show it should just get fresh results from the current line, and don't even consider last_messages

I agree, so should i remove the `yank_last_diagnostics` command (and the code that came with it)?